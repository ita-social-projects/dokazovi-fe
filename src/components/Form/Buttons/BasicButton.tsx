import { Button } from '@material-ui/core';
import clsx from 'clsx';
import React from 'react';
import { useStyles } from './styles/BasicButton.style';

interface IBasicButtonProps {
  type: 'accept' | 'sign';
  className?: string;
  disabled?: boolean;
  label?: string;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
}

export const BasicButton: React.FC<IBasicButtonProps> = ({
  type,
  className,
  disabled,
  label,
  onClick,
}) => {
  const classes = useStyles();
  let rootClass: string;
  switch (type) {
    case 'accept':
      rootClass = classes.basicAcceptButton;
      break;
    case 'sign':
      rootClass = classes.basicSignButton;
      break;
  }

  return (
    <Button
      className={clsx(rootClass, className)}
      type="submit"
      variant="contained"
      fullWidth
      disabled={disabled}
      onClick={onClick}
    >
      {label}
    </Button>
  );
};
