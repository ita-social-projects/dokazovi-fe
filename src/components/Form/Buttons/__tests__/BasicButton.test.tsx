import userEvent, { TargetElement } from '@testing-library/user-event';
import React from 'react';
import { render } from '@testing-library/react';
import { BasicButton } from '../BasicButton';

describe('Test for BasicButton component.', () => {
  it('Should renders a BasicButton with accept type.', () => {
    const { container } = render(<BasicButton type="accept" />);

    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveClass('BasicInput-basicAcceptButton-1');
  });

  it('Should renders a BasicButton with sign type.', () => {
    const { container } = render(<BasicButton type="sign" />);

    expect(container.firstChild).toHaveClass('BasicInput-basicSignButton-4');
  });

  it('Should handle onClick', () => {
    const handleClick = jest.fn();
    const { container } = render(
      <BasicButton type="accept" onClick={handleClick} />,
    );
    userEvent.click(container.firstChild as TargetElement);
    expect(handleClick).toBeCalled();
  });

  it('Should not be active when disabled.', () => {
    const { container } = render(<BasicButton type="accept" disabled />);

    const button = container.firstChild as TargetElement;
    expect(button).toBeDisabled();
  });

  it('Should be label in the button', () => {
    const label = 'text';
    const { container } = render(<BasicButton type="accept" label={label} />);
    const button = container.firstChild as TargetElement;
    expect(button).toHaveTextContent(label);
  });
});
