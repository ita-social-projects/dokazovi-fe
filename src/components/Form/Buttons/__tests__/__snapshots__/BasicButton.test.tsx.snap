// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for BasicButton component. Should renders a BasicButton with accept type. 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained BasicInput-basicAcceptButton-1 MuiButton-fullWidth"
    tabindex="0"
    type="submit"
  >
    <span
      class="MuiButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
